tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -493,7 +493,12 @@ policy_refresh_cagg_add(PG_FUNCTION_ARGS)
 			ereport(ERROR,
 					(errcode(ERRCODE_DUPLICATE_OBJECT),
 					 errmsg("continuous aggregate policy already exists for \"%s\"",
-							get_rel_name(cagg_oid))));
+							get_rel_name(cagg_oid)),
+					 errdetail("Only one continuous aggregate policy can be "
+							   "created per continuous aggregate and the continuous aggregate "
+							   "\"%s\" already has one policy with job id %d.",
+							   get_rel_name(cagg_oid),
+							   ((BgwJob *) linitial(jobs))->fd.id)));
```

Comment:
The detailed message is overly long and repeats what is already said in the main message. I also think it is implicitly understood that only one policy is allowed given the main message. Still, adding the job_id might be useful, but I would make the detailed message a lot snappier:

```suggestion
					 errdetail("The continuous aggregate policy for \"%s\" has job id %d.",
							   get_rel_name(cagg_oid),
							   ((BgwJob *) linitial(jobs))->fd.id)));
```

Reply:
@erimatnor It is unclear to some users that your message implies that only one policy is allowed. It is necessary to be clear and specific that only one policy is allowed.

Reply:
The I would suggest something like this to avoid a lot of repetition:

```suggestion
					 errdetail("Only one policy can be created per continuous aggregate and a policy with job id %d already exists for \"%s\".",
							   ((BgwJob *) linitial(jobs))->fd.id,
							   get_rel_name(cagg_oid))));
```

Reply:
I utilized part of your suggestion. I still think it is useful to be explicit that the policy is continuous aggregate, since it is possible to add retention policies to continuous aggregates too.